videos/videos.js

```diff
@@ -87,71 +87,64 @@ router.post("/", restricted, async (req, res) => {
   };
 
   try {
-        // Start with request to DS server, should get a 200 immediately if working
-        // Then go to rest of functions
-        // Potentially explore 
-
-        const song = await Songs.add(songObject);
-        // console.log(song);
-
-        const videoObjectComplete = { ...videoObject, song_id: song };
-
-        // console.log(videoObjectComplete);
-
-
-
-        // console.log(videoObjectComplete);
-        // we want song to return its id
-        // then we'll destructure or spread some object to add that in
-        // then we'll 'add' that object to videos.add
-        const video = await Videos.add(videoObjectComplete);
-
-        const videoId = uuid.v4();
-
-        axios
-          .post(
-            "http://sample.eba-5jeurmbw.us-east-1.elasticbeanstalk.com/entry",
-            null,
-            {
-              params: {
-                preview: preview,
-                video_id: videoId,
-              },
+    if (!title_short) {
+      res.status(400).json({ message: "Missing title_short!" });
+    } else {
+      if (!preview) {
+        res.status(400).json({ message: "Missing preview!" });
+      } else {
+        if (!artist) {
+          res.status(400).json({ message: "Missing artist!" });
+        } else {
+          if (!deezer_id) {
+            res.status(400).json({ message: "Missing deezer_id!" });
+          } else {
+            if (!location) {
+              res.status(400).json({ message: "Missing location!" });
+            } else {
+              if (!video_title) {
+                res.status(400).json({ message: "Missing video_title!" });
+              } else {
+                if (!user_id) {
+                  res.status(400).json({ message: "Missing user_id!" });
+                } else {
+
+                  const song = await Songs.add(songObject);
+
+                  const videoObjectComplete = { ...videoObject, song_id: song };
+
+                  // we want song to return its id
+                  // then we'll destructure or spread some object to add that in
+                  // then we'll 'add' that object to videos.add
+                  const video = await Videos.add(videoObjectComplete);
+                  const videoId = uuid.v4();
+                  
+                  axios
+                    .post(
+                      "http://sample.eba-5jeurmbw.us-east-1.elasticbeanstalk.com/entry",
+                      null,
+                      {
+                        params: {
+                          preview: preview,
+                          video_id: videoId
```

Comment:
so this is where the uuid gets created for a"newly initiated but not yet completed" Video?
 It looks like we're going to send that new uuid along with the deezer preview to the ds backend so that ds knows where to put the finished product.
 do we return that uuid to the client also or does the client ping something else like the songid?

This looks sane just talking it out.

Reply:
Yep, you got it